ninja-core/src/main/java/ninja/RouterImpl.java

```diff
@@ -185,30 +188,44 @@ public void compileRoutes() {
         }
         List<Route> routesLocal = new ArrayList<>();
         
-        for (RouteBuilderImpl routeBuilder : allRouteBuilders) {
+        allRouteBuilders.forEach(routeBuilder -> {
             routesLocal.add(routeBuilder.buildRoute(injector));
-        }
+        });
         
         this.routes = ImmutableList.copyOf(routesLocal);
         
         // compile reverse routes for O(1) lookups
         this.reverseRoutes = new HashMap<>(this.routes.size());
         
-        for (Route route : this.routes) {
-            // its possible this route is a Result instead of a controller method
-            if (route.getControllerClass() != null) {
-                MethodReference controllerMethodRef
-                    = new MethodReference(
-                        route.getControllerClass(),
-                        route.getControllerMethod().getName());
-                
-                if (this.reverseRoutes.containsKey(controllerMethodRef)) {
-                    // the first one wins with reverse routing so we ignore this route
-                } else {
-                    this.reverseRoutes.put(controllerMethodRef, route);
+        this.routes.forEach(route -> {
+            Class<?> methodClass = route.getControllerClass();
+            
+            String methodName = Optional.ofNullable(route.getControllerMethod())
+                .map(Method::getName)
+                .orElse("");
```

Comment:
not sure if this is super-clean... returning ""... I guess it simply not be added to the this.reverseRoutes afaics... (I may be wrong though...)

Reply:
This is partially a legacy of accepting static Results in old versions.  Those are now wrapped in functional lambdas -- and we can require a method name now.  Will clean this up.